### ISSUE TITLE
Benefits of using this kernel

### ISSUE THREAD

**Issue by christianbundy** [raphael/linux-samus#217]
Oops, accidentally hit <Enter> while trying to type this issue. Sorry about that.

---

I've set up a more streamlined build system for Arch, which basically just copies whatever the main Arch package is doing and applies the config changes, but now I'm wondering whether this is actually useful/helpful. Is anyone still using this project out of necessity, or are most of us using it out of habit?

There are only two non-trivial changes to the kernel config that seem relevant:

- `CONFIG_MCORE2=y`
- `CONFIG_TOUCHSCREEN_ATMEL_MXT=y`

The former *may* have some slight performance improvements, and the latter *may* have slightly better debugging (?) capabilities, but I'm not sure either of those make a material difference from the mainline kernel (or Arch upstream). Simply put: is anyone still using this kernel? If so, I'd like to keep it up-to-date, but I'm starting to question whether/why this would be used now that Debian Jessie has reached EOL.

**Comment by raphael** [comment]
At this point I think this kernel might have outlived its usefulness. I have been switching back and forth between the official (Archlinux) kernel and this one (which I've kept up-to-date locally) but can't really see any significant differences.

**Comment by ehegnes** [comment]
I agree. Sorry I haven't ended up having the time to maintain this either, but I've experienced the same with respect to switching kernels.

Some of the config is still valuable. I use the scripts contained here daily, and the other configs when I reinstall my system.

**Comment by ehegnes** [comment]
@christianbundy `CONFIG_MCORE` is defining processor family-specific changes that are supposed to improve performance. I compile my kernel with `-march=native` over a predefined family. If you're concerned about performance, I recommended that. 

You're likely not missing out on much without that optimization.

Compiling `CONFIG_TOUCHSCREEN_ATMEL_MXT` into the kernel is likely not improving anything either. If you're looking to change flags or debug, it would even be a hindrance over using a module.

**Comment by christianbundy** [comment]
Is there anything outside of [`scripts/setup`](https://github.com/raphael/linux-samus/tree/master/scripts/setup) that anyone is still using? This project has been massively helpful and a great resource, but once this package is flagged out-of-date on the AUR I'm not sure it sounds like any of us (?) are interested in releasing future versions.

**Comment by drivenbyentropy** [comment]
My apologies for hijacking this old issue but I believe the following falls under the umbrella of this ticket. I will be happy to open a separate ticket but since this is more likely and issue with upstream I thought I would be more appropriate to discuss this here first.

I am still using this kernel because for me, hibernation and suspension (sleep) when closing the lid on the 5.x kernels (Arch linux) causes a full shutdown instead of the expected behavior with no visible errors in the logs. The same is true when initiating sleep using `systemctl suspend`. 

The system appears to go to sleep successfully, but then shuts down and reboots when opening the lid again.

```bash
[root@pixel matrix]# journalctl | grep -A 500 "Lid closed"
Nov 01 23:46:27 pixel systemd-logind[502]: Lid closed.
Nov 01 23:46:27 pixel org_kde_powerdevil[645]: powerdevil: Suspend session triggered with QMap(("Explicit", QVariant(bool, true))("Type", QVariant(uint, 1)))
Nov 01 23:46:27 pixel org_kde_powerdevil[645]: powerdevil: Suspend session triggered with QMap(("Explicit", QVariant(bool, true))("SkipFade", QVariant(bool, true))("Type", QVariant(uint, 1)))
Nov 01 23:46:27 pixel org_kde_powerdevil[645]: powerdevil: Starting Login1 suspend job
Nov 01 23:46:27 pixel kdeinit5[570]: bluedevil: About to suspend
Nov 01 23:46:27 pixel org_kde_powerdevil[645]: powerdevil: Pausing all media players before sleep
Nov 01 23:46:27 pixel ksmserver[598]: lock called
Nov 01 23:46:27 pixel ksmserver[598]: Lock window Id:  23068679
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 23068679
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 23068681
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 33554562
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 56623257
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720260
Nov 01 23:46:27 pixel kscreenlocker_greet[863]: Qt: Session management error: networkIdsList argument is NULL
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720264
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720272
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720277
Nov 01 23:46:27 pixel kscreenlocker_greet[863]: kf5.kconfig.core: couldn't lock local file
Nov 01 23:46:27 pixel kscreenlocker_greet[863]: file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:76:9: Unable to assign [undefined] to QStringList
Nov 01 23:46:27 pixel kscreenlocker_greet[863]: file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:75:9: Unable to assign [undefined] to int
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720279
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720281
Nov 01 23:46:27 pixel ksmserver[598]: MapNotify: 58720279
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 39846652
Nov 01 23:46:27 pixel ksmserver[598]: CreateNotify: 58720283
Nov 01 23:46:27 pixel ksmserver[598]: MapNotify: 23068679
Nov 01 23:46:27 pixel systemd[1]: Reached target Sleep.
Nov 01 23:46:27 pixel systemd[1]: Starting Suspend...
Nov 01 23:46:27 pixel kdeinit5[570]: ktp-kded-module: "screen-saver-away" presence change request: "away" ""
Nov 01 23:46:27 pixel kdeinit5[570]: ktp-kded-module: plugin queue activation: "away" ""
Nov 01 23:46:27 pixel systemd-sleep[873]: Suspending system...
Nov 01 23:46:27 pixel kernel: PM: suspend entry (deep)
-- Reboot --
Nov 01 23:47:15 pixel kernel: Linux version 5.3.8-arch1-1 (linux@archlinux) (gcc version 9.2.0 (GCC)) #1 SMP PREEMPT @1572357769
Nov 01 23:47:15 pixel kernel: Command line: BOOT_IMAGE=../vmlinuz-linux root=/dev/mapper/cryptroot rw cryptdevice=UUID=ed95a468-799f-45d5-9fd8-1e837786cf75:cryptroot video=1920x1275 initrd=../initramfs-linux.img
Nov 01 23:47:15 pixel kernel: KERNEL supported cpus:
Nov 01 23:47:15 pixel kernel:   Intel GenuineIntel
Nov 01 23:47:15 pixel kernel:   AMD AuthenticAMD
Nov 01 23:47:15 pixel kernel:   Hygon HygonGenuine
Nov 01 23:47:15 pixel kernel:   Centaur CentaurHauls
Nov 01 23:47:15 pixel kernel:   zhaoxin   Shanghai  
[...]
```

The samus kernel in turn works as expected (even though the light bar does not turn off as it used to in previous versions). 

There are similar reports of issues with very close symptoms [reported online](https://www.google.com/search?q=arch+shutdown+on+lid+closed+site:bbs.archlinux.org&sa=X&ved=2ahUKEwjB2bf22svlAhWJg-AKHckICE0QrQIoBDAAegQIAhAN&biw=1707&bih=1042), non of which provide a transferable solution. 

Is anyone experiencing this issue as well?

**Comment by christianbundy** [comment]
I'm experiencing this issue, I reported it in the forums here: https://bbs.archlinux.org/viewtopic.php?id=246716

Unfortunately the only recommendations were:

- Use `linux-lts` and hope the bug is fixed before it hits LTS.
- Bisect the kernel and figure out exactly which commit caused the problem.

FWIW, the problem started between 4.19.46 and 5.1.5. Maybe that narrows it down a bit? I've been meaning to take a look at this but I've been busy with other things.

Related: I've never had a functioning microphone with this device, anyone know the fix for *that*?

**Comment by drivenbyentropy** [comment]
@christianbundy, thanks for confirming, its good to know its not just me. 

In the arch forum you mentioned a BIOS update that you performed. Could you point me in the right direction how to take on this task?

As for the microphone, it is working without issues for me on the samus kernel. I ran the sound setup scripts and configured the input/output channels using `pavucontrol`.

Meanwhile, I will downgrade to 4.19.46 and report back. Long term however, it would be advantageous to find the upstream commit that causes this behavior and report it. Maybe we could join forces and figure this out? 

Cheers!

**Comment by christianbundy** [comment]
> Could you point me in the right direction how to take on this task?

Here's what I've been using: https://mrchromebox.tech/#fwscript

FWIW, I'm just using SeaBIOS because I've been avoiding switching to UEFI.

> I ran the sound setup scripts and configured the input/output channels using pavucontrol.

!!! I ran `pavucontrol` and saw that there was a tiny bit of output, and then I opened alsamixer and went to the capture settings and increased everything to 100%. It works! Thanks for solving my years of microphone struggles.

> Maybe we could join forces and figure this out?

I'll try to find some time today for a bisect and report back with any info I have. :+1: 

---

## Arch Package Bisect

| linux version | arch package commit                                   |  status |
|---------------|------------------------------------------|---------|
| 5.0.13        | 62fb9609befa19b14c522c311b4dc25b30c827c5 | good  |  
| 5.0.13        | 14cc58e99e142aca0e0813785c82c3ea9e75cf9b | good |
| **5.1**            | **d706a3e8a2deddc677303b029c5df81378948f75** | **bad**     |
| 5.1.1         | 48d3d8ee2ff3750e147645c36b95ce7311a19455 | bad     |
| 5.1.4         | d5b05a2aac4c80866e58e6d3ba03b3bded4e1962 | bad     |
| 5.1.9         | 48f7a4ad6b2436e5fc17cd20e0070bf6da672b93 | bad     |

I finished bisecting the Arch VCS but that doesn't tell me which commit from the Linux VCS caused the problem. I'll tinker with it a bit and see if I can easily bisect the kernel without giving up the sweet `makepkg `integration. Would love tips if anyone knows The Right Way to do this.

Current technique: manage the bisect manually in `src/archlinux-linux` and prevent `makepkg` from touching anything with:

## Kernel Bisect

```diff
diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 93255d2..d3e2e65 100644
--- a/trunk/PKGBUILD
+++ b/trunk/PKGBUILD
@@ -14,7 +14,7 @@ makedepends=(xmlto kmod inetutils bc libelf git)
 options=('!strip')
 _srcname=archlinux-linux
 source=(
-  "$_srcname::git+https://git.archlinux.org/linux.git?signed#tag=v$_srcver"
+  #"$_srcname::git+https://git.archlinux.org/linux.git?signed#tag=v$_srcver"
   config         # the main kernel config file
   60-linux.hook  # pacman hook for depmod
   90-linux.hook  # pacman hook for initramfs regeneration
@@ -25,7 +25,7 @@ validpgpkeys=(
   '647F28654894E3BD457199BE38DBBDC86092693E'  # Greg Kroah-Hartman
   '8218F88849AAC522E94CF470A5E9288C4FA415FA'  # Jan Alexander Steffens (heftig)
 )
-sha256sums=('SKIP'
+sha256sums=(#'SKIP'
             'd1ad8403bfc9681bddf42300b1b51d8975506b91e791e341f839797513208eff'
             'ae2e95db94ef7176207c690224169594d49445e04249d2499e9d2fbc117a0b21'
             'c043f3033bb781e2688794a59f6d1f7ed49ef9b13eb77ff9a425df33a244a636'
@@ -61,6 +61,7 @@ prepare() {
 
 build() {
   cd $_srcname
+  make localmodconfig
   make bzImage modules
 }
 ```

(I think `make localmodconfig` speeds up the build. Not 100% sure.)

Here's my `git bisect log` so far:

```
git bisect start
# good: [6ee2c3f93d50b2a9aa7bee32bf16b924f2d98596] Arch Linux kernel v5.0.13-arch1
git bisect good 6ee2c3f93d50b2a9aa7bee32bf16b924f2d98596
# bad: [131e410f795e5d9bbbc075ddec0aa7af4ac75038] Arch Linux kernel v5.1-arch1
git bisect bad 131e410f795e5d9bbbc075ddec0aa7af4ac75038
# good: [1c163f4c7b3f621efff9b28a47abb36f7378d783] Linux 5.0
git bisect good 1c163f4c7b3f621efff9b28a47abb36f7378d783
# good: [b5dd0c658c31b469ccff1b637e5124851e7a4a1c] Merge branch 'akpm' (patches from Andrew)
git bisect good b5dd0c658c31b469ccff1b637e5124851e7a4a1c
# bad: [4f0237062ca70c8e34e16e518aee4b84c30d1832] Merge branch 'for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/dtor/input
git bisect bad 4f0237062ca70c8e34e16e518aee4b84c30d1832
# good: [80201fe175cbf7f3e372f53eba0a881a702ad926] Merge tag 'for-5.1/block-20190302' of git://git.kernel.dk/linux-block
git bisect good 80201fe175cbf7f3e372f53eba0a881a702ad926
# good: [a50243b1ddcdd766d0d17fbfeeb1a22e62fdc461] Merge tag 'for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/rdma/rdma
git bisect good a50243b1ddcdd766d0d17fbfeeb1a22e62fdc461
# good: [b7a7d1c1ec688104fdc922568c26395a756f616d] Merge tag 'dma-mapping-5.1' of git://git.infradead.org/users/hch/dma-mapping
git bisect good b7a7d1c1ec688104fdc922568c26395a756f616d
# good: [c3665a6be5de16cf6670a00003642114c44d8a70] Merge branch 'next-integrity' of git://git.kernel.org/pub/scm/linux/kernel/git/jmorris/linux-security
git bisect good c3665a6be5de16cf6670a00003642114c44d8a70
# bad: [ffd602eb4693bbb49b301fa059b109bbdebf9524] Merge tag 'kbuild-v5.1' of git://git.kernel.org/pub/scm/linux/kernel/git/masahiroy/linux-kbuild
git bisect bad ffd602eb4693bbb49b301fa059b109bbdebf9524
# good: [9250d20e9ecedab6aa331a127fbfc1272383ed72] kbuild: remove scripts/basic/% build target
git bisect good 9250d20e9ecedab6aa331a127fbfc1272383ed72
# good: [29b47ce987591254602cd7b69acd0eecc62f0a70] tpm: move TPM space code out of tpm_transmit()
git bisect good 29b47ce987591254602cd7b69acd0eecc62f0a70
# bad: [901615cb916dc955fb7bda4e34402bf263532e4a] tpm: move tpm_chip definition to include/linux/tpm.h
git bisect bad 901615cb916dc955fb7bda4e34402bf263532e4a
# bad: [a3fbfae82b4cb3ff9928e29f34c64d0507cad874] tpm: take TPM chip power gating out of tpm_transmit()
git bisect bad a3fbfae82b4cb3ff9928e29f34c64d0507cad874
# good: [2677ca98ae377517930c183248221f69f771c921] tpm: use tpm_try_get_ops() in tpm-sysfs.c.
git bisect good 2677ca98ae377517930c183248221f69f771c921
# good: [719b7d81f2048477f89f7303e2b4ddec8197e6e3] tpm: introduce tpm_chip_start() and tpm_chip_stop()
git bisect good 719b7d81f2048477f89f7303e2b4ddec8197e6e3
# first bad commit: [a3fbfae82b4cb3ff9928e29f34c64d0507cad874] tpm: take TPM chip power gating out of tpm_transmit()
```

## Notes

I think I've found a pattern in the TPM logs (`journalctl -b | grep -i tpm`). We usually see this during boot:

```
kernel: tpm_tis 00:08: 1.2 TPM (device-id 0xB, rev-id 16)
kernel: tpm tpm0: [Firmware Bug]: TPM interrupt not working, polling instead
```

On kernels with this bug, we instead see:

```
kernel: tpm_tis 00:08: 1.2 TPM (device-id 0xB, rev-id 16)
kernel: tpm tpm0: tpm_try_transmit: send(): error -5
kernel: tpm tpm0: A TPM error (-5) occurred attempting to determine the timeouts
kernel: tpm_tis 00:08: Could not get TPM timeouts and durations
```

I'm resisting the urge to investigate any workarounds until I know which commit caused the regression, but it feels good to see some relevant info in the logs.

**Comment by drivenbyentropy** [comment]
Nice, glad to hear the Mic is working now. 

I will join you in bisecting later today (need to run a couple of errands first).  :+1:

**Comment by drivenbyentropy** [comment]
Thank you for this!

> Would love tips if anyone knows The Right Way to do this.

Do you think it would be worth posting the progress and question to the arch forum post (as old as it might be)?

**Comment by christianbundy** [comment]
Found the first bad commit! https://github.com/torvalds/linux/commit/a3fbfae82b4cb3ff9928e29f34c64d0507cad874

I've replied in the [Arch Linux forum thread](https://bbs.archlinux.org/viewtopic.php?pid=1871717#p1871717), should we move the discussion over there?

**Comment by drivenbyentropy** [comment]
This is great news, thank you @christianbundy for tracking down the commit and for opening the [bug report](https://bugzilla.kernel.org/show_bug.cgi?id=205409). I will try to reproduce your findings and confirm on the forums.

**Comment by drivenbyentropy** [comment]
It took me a while but I was able to replicate your findings. Kernel ```linux-git-5.0rc3.r81.g719b7d81f204``` (corresponding to the last good commit of your bisect) works as expected, however future versions do not. Lets hope the bug report will get some attention soon.

Thanks again for finding this, I can now close my Pixel again without fear of loss of data :)

PS: Is the light bar at the back of the screen working for you with the custom build? I seem to have lost  it while configuring the kernel.

**Comment by christianbundy** [comment]
Happy to hear that. I think maybe I'll post to the mailing list in a few days if I don't hear anything,

> PS: Is the light bar at the back of the screen working for you with the custom build? I seem to have lost it while configuring the kernel.

Yep, I don't think this is something I've ever had problems. I *thought* it was a firmware thing but maybe not?

**Comment by drivenbyentropy** [comment]
Strange, the backlight came back after a couple of reboots.

Since we seem to have a very similar setup on our Pixels, I was wondering if you have experimented with hibernation? I have a LUKS encrypted root partition with a swapfile on that partition and set up hibernation following [this article](https://wiki.archlinux.org/index.php/Power_management/Suspend_and_hibernate#Hibernation_into_swap_file). When executing ```systemctl hibernate```, the system goes into hibernation successfully and boots back into its hibernated state after entering the decryption password. However, the trackpad and touchscreen stops working and only comes back to life after rebooting the system (see log below). Have you encountered this issue as well?

```bash
Nov 05 22:16:45 pixel systemd[1]: Starting Hibernate...
Nov 05 22:16:45 pixel kdeinit5[600]: ktp-kded-module: "screen-saver-away" presence change request: "away" ""
Nov 05 22:16:45 pixel kdeinit5[600]: ktp-kded-module: plugin queue activation: "away" ""
Nov 05 22:16:45 pixel systemd-sleep[894]: Suspending system...
Nov 05 22:16:45 pixel kernel: PM: hibernation entry
Nov 05 22:16:45 pixel kernel: PM: Syncing filesystems ... 
Nov 05 22:18:02 pixel kernel: PM: done.
Nov 05 22:18:02 pixel kernel: Freezing user space processes ... (elapsed 0.002 seconds) done.
Nov 05 22:18:02 pixel kernel: OOM killer disabled.
Nov 05 22:18:02 pixel kernel: PM: Marking nosave pages: [mem 0x00000000-0x00000fff]
Nov 05 22:18:02 pixel kernel: PM: Marking nosave pages: [mem 0x0009f000-0x000fffff]
Nov 05 22:18:02 pixel kernel: PM: Marking nosave pages: [mem 0x7ce28000-0xffffffff]
Nov 05 22:18:02 pixel kernel: PM: Basic memory bitmaps created
Nov 05 22:18:02 pixel kernel: PM: Preallocating image memory... done (allocated 546922 pages)
Nov 05 22:18:02 pixel kernel: PM: Allocated 2187688 kbytes in 0.32 seconds (6836.52 MB/s)
Nov 05 22:18:02 pixel kernel: Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
Nov 05 22:18:02 pixel kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Nov 05 22:18:02 pixel kernel: wlp1s0: deauthenticating from 12:80:63:33:75:d9 by local choice (Reason: 3=DEAUTH_LEAVING)
Nov 05 22:18:02 pixel kernel: ACPI: EC: interrupt blocked
Nov 05 22:18:02 pixel kernel: ACPI: Preparing to enter system sleep state S4
Nov 05 22:18:02 pixel kernel: ACPI: EC: event blocked
Nov 05 22:18:02 pixel kernel: ACPI: EC: EC stopped
Nov 05 22:18:02 pixel kernel: PM: Saving platform NVS memory
Nov 05 22:18:02 pixel kernel: Disabling non-boot CPUs ...
Nov 05 22:18:02 pixel kernel: smpboot: CPU 1 is now offline
Nov 05 22:18:02 pixel kernel: smpboot: CPU 2 is now offline
Nov 05 22:18:02 pixel kernel: smpboot: CPU 3 is now offline
Nov 05 22:18:02 pixel kernel: PM: Creating hibernation image:
Nov 05 22:18:02 pixel kernel: PM: Need to copy 544475 pages
Nov 05 22:18:02 pixel kernel: PM: Normal pages needed: 544475 + 1024, available pages: 3632567
Nov 05 22:18:02 pixel kernel: PM: Restoring platform NVS memory
Nov 05 22:18:02 pixel kernel: ACPI: EC: EC started
Nov 05 22:18:02 pixel kernel: Enabling non-boot CPUs ...
Nov 05 22:18:02 pixel kernel: x86: Booting SMP configuration:
Nov 05 22:18:02 pixel kernel: smpboot: Booting Node 0 Processor 1 APIC 0x1
Nov 05 22:18:02 pixel kernel:  cache: parent cpu1 should not be sleeping
Nov 05 22:18:02 pixel kernel: CPU1 is up
Nov 05 22:18:02 pixel kernel: smpboot: Booting Node 0 Processor 2 APIC 0x3
Nov 05 22:18:02 pixel kernel:  cache: parent cpu2 should not be sleeping
Nov 05 22:18:02 pixel kernel: CPU2 is up
Nov 05 22:18:02 pixel kernel: smpboot: Booting Node 0 Processor 3 APIC 0x2
Nov 05 22:18:02 pixel kernel:  cache: parent cpu3 should not be sleeping
Nov 05 22:18:02 pixel kernel: CPU3 is up
Nov 05 22:18:02 pixel kernel: ACPI: Waking up from system sleep state S4
Nov 05 22:18:02 pixel kernel: ACPI: EC: interrupt unblocked
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3432:00: timeout in disabling adapter
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout in disabling adapter
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: usb usb1: root hub lost power or was reset
Nov 05 22:18:02 pixel kernel: usb usb2: root hub lost power or was reset
Nov 05 22:18:02 pixel kernel: ACPI: EC: event unblocked
Nov 05 22:18:02 pixel kernel: ACPI: button: The lid device is not compliant to SW_LID.
Nov 05 22:18:02 pixel kernel: sd 0:0:0:0: [sda] Starting disk
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3432:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0000:01: __mxt_write_reg: i2c send failed (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_read_reg: i2c transfer failed (-110)
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: Failed to read T44 and T5 (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3432:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0000:01: __mxt_write_reg: i2c send failed (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
Nov 05 22:18:02 pixel kernel: atmel_mxt_ts i2c-ATML0001:01: __mxt_write_reg: i2c send failed (-110)
Nov 05 22:18:02 pixel kernel: i2c_designware INT3433:00: timeout waiting for bus ready
[...]
```

**Comment by christianbundy** [comment]
I think the main question here was answered (thanks again!). Closing and moving discussion to new issue.
